truffle/src/main/java/org/jruby/truffle/nodes/core/CoreMethodNodeManager.java

```diff
@@ -210,6 +210,10 @@ private static RubyRootNode makeGenericMethod(RubyContext context, MethodDetails
         final CheckArityNode checkArity = new CheckArityNode(context, sourceSection, arity);
         RubyNode sequence = SequenceNode.sequence(context, sourceSection, checkArity, methodNode);
 
+        if (methodDetails.getMethodAnnotation().returnsEnumeratorIfNoBlock()) {
+            sequence = new ReturnEnumeratorIfNoBlockNode(methodDetails.getMethodAnnotation().names()[0], sequence);
```

Comment:
Using the first name here might not be correct. As it currently is, if you call `collect` without a block it will do `to_enum :map`, which isn't always going to be correct. We can handle that later, but can you put a comment in for now?